lib/web_catalog/post_processors.es6.js

```diff
@@ -192,23 +229,53 @@ foam.CLASS({
 
       // Allow post-processor to run on object graphs that don't define the
       // input interface.
-      var fromId = og.lookup(`window.${this.fromInterfaceName}`);
+      const fromId = og.lookup(this.fromInterfaceName);
       if (!fromId) return;
-      var fromProtoId = og.lookup('prototype', fromId);
+      // Prototype is either:
+      // (1) <fromId-as-function>.prototype, OR
+      // (2) <fromId-as-object>.
+      const fromProtoId = og.getFunctionName(fromId) ?
+          og.lookup('prototype', fromId) : fromId;
       if (!fromProtoId) return;
-      var toProtoId = og.getPrototype(fromProtoId);
+      const toProtoId = og.getPrototype(fromProtoId);
       if (!toProtoId) return;
 
-      var keyParts = og.getShortestKey(toProtoId).split('.');
-      foam.assert(
-          keyParts[0] === 'window' && keyParts[2] === 'prototype' &&
-              keyParts.length === 3,
-          `Expected CopyToPrototypeProcessor prototype to have string key of
-          window.<prototype ctor name>.prototype.`);
-      var toName = keyParts[1];
+      const interfaceNames = og.getKeys(toProtoId)
+            .filter(key => key.endsWith('.prototype'))
+            .map(key => {
+              const parts = key.split('.');
+              return parts[parts.length - 2];
+            });
+      foam.assert(interfaceNames.length > 0,
+                  `Expected CopyToPrototypeProcessor prototype to have string
+                       key(s) of ...<prototype ctor name>.prototype.`);
 
-      apiCatalog[toName] = apiCatalog[toName]
-          .concat(apiCatalog[this.fromInterfaceName]);
+      const newAPIs = apiCatalog[this.fromInterfaceName];
```

Comment:
The merge-with-shadowing logic here and above are the same. I don't know if it makes more sense to construct a CopyToInterfaceProcessor for each discovered interface and run it instead of duplicating the two lines, since the merge logic is unlikely to change.

Reply:
I'm not sure I follow. Do you mean construct a `CopyToInterfaceProcessor(foo, foo.__proto__)` instead of `CopyToPrototypeProcessor` at all?

Reply:
No, I meant use CopyToInterfaceProcessor to do the copy here. This processor is a little bit more complicated because it scans for destinations first, but once it has them it's just repeating the copying operation as in CopyToInterfaceProcessor. But as I said, it avoids repeating about two lines of logic, so it might not make sense.